src/service/credit_service.py

```
@@ -0,0 +1,320 @@
+            return 10
+        if event_type == CreditEventType.CODE_EXECUTION.value:
+            return 5
+
+        # 기능별 비용 계산 (난이도 적용)
+        base_cost = FEATURE_BASE_COST.get(feature_name, 5)
+        multiplier = DIFFICULTY_MULTIPLIER.get(difficulty.lower() if difficulty else "easy", 1.0)
+        return int(base_cost * multiplier + 0.5)  # 반올림
+
+    async def get_balance(
+        self,
+        user_id: str,
```

Ruff (0.14.14)

[warning] Unused method argument: `user_id`

(ARG002)